### PR TITLE
Complete walkthroughs/week-3-server/step-4

### PR DESCRIPTION
**Pull request by tayyaba17**
Some hard-code remains and will be updated in step 5 when I set up Datastore. There is also some non-functional code present that I will remove within step 5, just in case I need to when implementing those steps. Let me know if anything is confusing, thank you for your help!

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -29,15 +30,37 @@
...
+    response.getWriter().println(Arrays.toString(words));
+
+   
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:57]: Nice! Good to use small utility function for things like this!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -29,15 +30,37 @@
...
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
+
+    // Respond with the result.
+    response.setContentType("text/html;");
+    response.getWriter().println(Arrays.toString(words));
+
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Nit: it's good not to leave extra empty lines here:)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -29,15 +30,37 @@
...
   }
 
-  private Gson gson;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: From this stackoverflow search[1], someone also suggests using

String[] words = s.split("\\W+");

as a more general solution for ASCII. Yours works as fine for words split by comma or space, so it's your call.

[1] https://stackoverflow.com/questions/11726023/split-string-into-individual-words-java

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -73,53 +73,62 @@ <h1>Makeup Gallery</h1>
...
+    </div>
 
-  <div class="slides-fade">
-    <img src="images/GP5.jpg" alt="My guinea pig, Butter, running towards the camera" 
-    width="550" 
-    height="550">
-    <div class="text">My guinea pig, Butter, running towards the camera</div>
-  </div>
+    <div class="slides-fade">
+       <img src="images/GP5.jpg" alt="My guinea pig, Butter, running towards the camera"
```
Comment (peizhaoou) [inline portfolio/src/main/webapp/index.html:106]: Nit: Should we only use 2-space indentation?

Comment (martiKallas) [reply portfolio/src/main/webapp/index.html:106]: +1, it looks like some of the indents got mixed up during the change.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -72,54 +72,54 @@ <h1>Makeup Gallery</h1>
       <img src="images/MU9.JPG" alt="A look inspired by BTS's music video, Euphoria"
        width="650" height="650">
     </div>
-  </body>
+
```
Comment (peizhaoou) [inline portfolio/src/main/webapp/index.html:75]: Thanks for fixing the indentation.

**Pull request merged**